compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -4075,6 +4127,20 @@ public void visit(BLangConstant constant, AnalyzerData data) {
         constantAnalyzer.visit(constant);
     }
 
+    private boolean isNodeKindAllowedForConstants(BLangExpression expression, BLangConstant constant) {
+        if ((expression.getKind() == NodeKind.UNARY_EXPR &&
+                ((BLangUnaryExpr) constant.expr).expr.getKind() != NodeKind.NUMERIC_LITERAL)  &&
+                (!OperatorKind.SUB.equals(((BLangUnaryExpr) constant.expr).operator) ||
+                !OperatorKind.ADD.equals(((BLangUnaryExpr) constant.expr).operator))) {
+            return constant.typeNode == null;
+        }
+        if (!(expression.getKind() == LITERAL || expression.getKind() == NUMERIC_LITERAL) &&
+                expression.getKind() != NodeKind.UNARY_EXPR) {
+            return constant.typeNode == null;
```

Comment:
Didn't we add util methods for these checks?

Reply:
This check is slightly different from the check in util method... since this is for constants.. like with leaving out the check for group expr because we have only special cased constants for unary expr and so on...